src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -3386,7 +3386,14 @@ QString QgsProcessingParameterFileDestination::valueAsPythonString( const QVaria
 
 QgsProcessingOutputDefinition *QgsProcessingParameterFileDestination::toOutputDefinition() const
 {
-  return nullptr;
+  if ( !mFileFilter.isEmpty() and mFileFilter.contains( QStringLiteral( "html" ), Qt::CaseInsensitive ) )
```

Comment:
and -> &&

Reply:
maybe this should be just "htm" instead of "html"? We're testing for contains, so both would be caught then.

Reply:
Fixed, thanks